test/jdk/java/nio/channels/Selector/java.base/java/net/InetSocketAddress.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.net;
+
+
+import java.util.Locale;
+
+// Patched implementation only meant to be used in certain tests
+public class InetSocketAddress extends SocketAddress {
+
+    @java.io.Serial
+    private static final long serialVersionUID = 5076001401234631237L;
+
+    public static final ThreadLocal<Boolean> INTRODUCE_DELAY = new ThreadLocal<>() {
+        @Override
+        protected Boolean initialValue() {
+            return false;
+        }
+    };
```

Comment:
A simpler way to do this is to just add a setDelay method or even a constructor that allows the delay to be specified when creating the object.  That will allow the ThredLocal to go away and make the test a bit simpler.

Reply:
The use of ThreadLocal was to add fine grained delays to avoid delaying all the places where InetSocketAddress gets used and increasing the test execution time. I will remove the ThreadLocal and see how big an impact it has on test durations.

Reply:
Okay, whatever is easiest. My comment was really to say that a method or constructor would catch any issues at compile-time and avoid any implicit parameters. So what is easiest, this is hard test to write.

My only significant comment is about the execution time and whether we should dial it down a bit to reduce the execution time. It doesn't matter if the test doesn't reproduce every time because the test will run enough times on different systems to exercise the code path.